pyvista/utilities/fileio.py

```diff
@@ -384,48 +119,72 @@ def read(filename, attrs=None, force_ext=None, file_format=None, progress_bar=Fa
     if not os.path.isfile(filename):
         raise FileNotFoundError(f'File ({filename}) not found')
 
-    ext = _get_ext_force(filename, force_ext)
-
     # Read file using meshio.read if file_format is present
     if file_format:
         return read_meshio(filename, file_format)
 
-    # From the extension, decide which reader to use
-    if attrs is not None:
-        reader = get_vtk_reader(filename, force_ext=ext)
-        return standard_reader_routine(reader, filename, attrs=attrs, progress_bar=progress_bar)
-    elif ext in ['.e', '.exo']:
+    ext = _get_ext_force(filename, force_ext)
+    if ext in ['.e', '.exo']:
         return read_exodus(filename)
-    elif ext in ['.vtk']:
-        # Attempt to use the legacy reader...
-        return read_legacy(filename, progress_bar=progress_bar)
-    else:
-        # Attempt find a reader in the readers mapping
+
+    try:
+        reader = pyvista.get_reader(filename, force_ext)
+    except ValueError:
         try:
-            reader = get_vtk_reader(filename, force_ext=ext)
-            return standard_reader_routine(reader, filename, progress_bar=progress_bar)
-        except KeyError:
-            # Don't fall back to meshio if using `force_ext`, which is really
-            # just intended to be used with the native PyVista readers
-            if force_ext is not None:
-                from meshio._exceptions import ReadError
-
-                raise ReadError
-            # Attempt read with meshio
+            from meshio._exceptions import ReadError
+
             try:
-                from meshio._exceptions import ReadError
-
-                try:
-                    return read_meshio(filename)
-                except ReadError:
-                    pass
-            except SyntaxError:
-                # https://github.com/pyvista/pyvista/pull/495
+                return read_meshio(filename)
+            except ReadError:
                 pass
+        except SyntaxError:
```

Comment:
This seems weird and the commit in #495 that adds this references Python 3.5.  Can we remove this except?

Reply:
Good catch, yes please! Seems like `meshio` was using f-strings while <3.6 was supported in PyVista. We don't even have 3.6 now, this should go.

Reply:
Good, this allows for more cleanup here (and allows a higher % of coverage of the changes).